configure.ac

```diff
@@ -354,38 +354,42 @@ if test x"$linux" = x"yes"; then
 fi
 
 #
-# Check for res_nquery in libresolv. There are several variations. On OSX
+# Check for res_nquery or res_query in libresolv. There are several variations. On OSX
 # res_nquery is a macro defined in resolv.h, so the typical AC_CHECK_LIB
-# doesn't work. On FreeBSD res_nquery can be found in libc. The required
-# library for linking (if any) is defined RESOLV_LIBS.
+# doesn't work. On FreeBSD res_nquery can be found in libc. On Alpine res_query can be
+# found in libc. The required library for linking (if any) is defined RESOLV_LIBS.
 #
 AC_CHECK_HEADER([resolv.h], [], [], [#include <netinet/in.h>])
 
 AC_DEFUN([CHECK_RESOLV_LIBS], [{
-  AC_MSG_CHECKING([if res_nquery will link (LIBS=$1)])
+  AC_MSG_CHECKING([if $1 will link (LIBS=$3)])
   saved_LIBS="${LIBS}"
-  LIBS="$1"
+  LIBS="$3"
   AC_LINK_IFELSE([AC_LANG_PROGRAM([[
         #include <netinet/in.h>
         #include <resolv.h>
-      ]], [[res_nquery(0, 0, 0, 0, 0, 0);]])],
+      ]], [[$1$2;]])],
```

Comment:
@tchaikov Don't think so? res_query and res_nquery have different number of arguments.

res_nquery(res_state statep, const char *dname, int class, int type, unsigned char *answer, int anslen)
res_query(const char *dname, int class, int type, unsigned char *answer, int anslen)

http://man7.org/linux/man-pages/man3/resolver.3.html